shm/map.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <vector>
+#include <memory>
+#include <functional>
+#include <algorithm>
+
+class Map {
+public:
+    Map();
+    Island* const getPosition();
+    Island* getIsland(const Island::Coordinates& coordinate);
+private:
+    const int maxWidth_ = 99;
+    const int maxHeight_ = 99;
+    const int islandsQuantity_ = 10;
+    std::vector<Island&> islands_;
```

Comment:
Vecotr of non const reference? I guess you don't compile it :) Change to std::reference_wrapper or pointer